.github/workflows/smoketest.yml

```diff
@@ -31,6 +31,11 @@ name: smoketest
 env:
   cacheversion: 3
   VCPKG_BINARY_SOURCES: "clear;nuget,GitHub,read"
+  OS_DEPENDENCIES: "bison flex build-essential binutils-dev curl lsb-release libldap2-dev libcppunit-dev libicu-dev
+    libxslt1-dev zlib1g-dev libboost-regex-dev libarchive-dev python3-dev default-jdk libapr1-dev libaprutil1-dev
+    libiberty-dev libhiredis-dev libtbb-dev libxalan-c-dev libnuma-dev libevent-dev libatlas-base-dev libblas-dev
+    python3-dev default-libmysqlclient-dev libsqlite3-dev r-base-dev r-cran-rcpp r-cran-rinside r-cran-inline
+    libmemcached-dev pkg-config libtool autotools-dev automake libssl-dev git libcurl4-openssl-dev cmake libbsd-dev"
```

Comment:
A lot of these will no longer be needed.  From memory the following can be removed:
libldap2-dev libcppunit-dev libicu-dev libxslt1-dev zlib1g-dev libboost-regex-dev libarchive-dev libapr1-dev libaprutil1-dev  libiberty-dev libhiredis-dev libtbb-dev libxalan-c-dev libnuma-dev libevent-dev libatlas-base-dev libblas-dev default-libmysqlclient-dev libsqlite3-dev  libmemcached-dev libssl-dev libcurl4-openssl-dev libbsd-dev

Reply:
@GordonSmith - Is that true only for >= 8.12 ?
If it is, then it suggest this should be conditional on base_ref version detected,
or perhaps more simply and generically (for any base ref branch) - detect whether a branch based on vcpkg changes.

And should the build wiki be updated?

Reply:
For 8.10 the list can also be pruned (but not to the same extent) - but this PR is targeting 8.12.x so will only be used by actions targeting this branch?  IOW a PR for 8.10.x will still trigger the smoketest from that branch?

Yes the Wiki AND BUILD.md doc could both do with an overhaul...

Reply:
you're right, changes will only effect future >= 8.12.x PR's as previous branches will not have these action changes in them.

So I'll prune the dependencies here.

Reply:
@GordonSmith - please see new commit.